.github/workflows/rust.yml

```diff
@@ -419,3 +419,38 @@ jobs:
 
       - if: "failure()"
         uses: "andymckay/cancel-action@271cfbfa11ca9222f7be99a47e8f929574549e0a" # v0.4
+
+  no-std-tests:
+    name: "Test (no_std)"
+    runs-on: ubuntu-latest
+    needs: [machete, docs]
+    timeout-minutes: 30
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      # Note: needs nighly toolchain because otherwise we cannot define custom lang-items.
+      - name: Install Rust nightly toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+
+      - name: Add the `thumbv7em-none-eabi` target
+        run: |
+          rustup target add `thumbv7em-none-eabi`
```

Comment:
We can use a single line syntax for this `run: rustup target add `thumbv7em-none-eabi`

Reply:
you could also add it to `target` in the install rustc GHA:

```
      - name: Install Rust nightly toolchain
        uses: actions-rs/toolchain@v1
        with:
          profile: minimal
          toolchain: nightly
          override: true
          target: thumbv7em-none-eabi
```